### PR TITLE
Initial drop of container interconnect section

### PR DESCRIPTION
**Pull request by imcleod**
I'm PRing this not because I think it is fully baked (It isn't even linked into the main document) but because I could use some initial feedback on style and format (and I have placeholders for diagrams, which I'll need some help with).

### CODE REVIEW

File: planning/planning_scenarios_distributed_db.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+By their very nature, distributed application components need to communicate with one another.  
+Container technologies encourage developers to make these interconnection points explicit and 
+provide a number of mechanisms to coordinate or otherwise enable communication between containers.
+
+Consider the database example in the previous section.  Once we have established persistent 
+storage for the database server, we also need to consider how database clients will connect to it.
+In nearly all cases these connections will occur through a socket, ever over the network or locally
+via UNIX domain socket special file.
+
+(Diagram placeholder - Block for running container, inset for listening port on top of container
```
Comment (baude) [inline planning/planning_scenarios_distributed_db.adoc:14]: Want to have me draw you a picture so when this is merged, it's all inclusive?

File: planning/planning_scenarios_distributed_db.adoc [hunk 2]
```diff
@@ -0,0 +1,59 @@
+=== Container Interconnection: Distributed Database Server
+
+Introduce the basic environment and began to foreshadow the perceieved complications
+
+==== Database server
```
Comment (baude) [inline planning/planning_scenarios_distributed_db.adoc]: ==== Traditional database server/environment?

File: planning/planning_scenarios_distributed_db.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
+=== Container Interconnection: Distributed Database Server
+
+Introduce the basic environment and began to foreshadow the perceieved complications
```
Comment (baude) [inline planning/planning_scenarios_distributed_db.adoc]: Perhaps a few sentences to kick off the topic?

File: planning/planning_scenarios_distributed_db.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+==== Container environment - Docker
+
+Once an interconnection mechanism has been established a container developer must ensure 
+that service containers are properly configured to allow access to these connections.
+
+For network connections, Docker provides a simple directive in the Dockerfile to expose a port from the 
+running container.  For example, to create a Postgres DB server container that listents on the default 
+Postgres port, you would add the following line:
+
+EXPOSE 5432
```
Comment (baude) [inline planning/planning_scenarios_distributed_db.adoc:35]: Did we discuss talking generally about port exposure and then pointing to the existing document (if it exists) for the implementation?  Same goes for bind mounts and so on.

File: planning/planning_scenarios_distributed_db.adoc (hunk 1 above)
Comment (imcleod) [inline planning/planning_scenarios_distributed_db.adoc:14]: Sure. :-)

File: planning/planning_scenarios_distributed_db.adoc (hunk 4 above)
Comment (imcleod) [inline planning/planning_scenarios_distributed_db.adoc:35]: We did indeed, apologies  - Those would be, I think, Network details in "4.8. Network Considerations" (which is currently empty in my branch) and Storage details in "4.9. Storage Considerations" (which is not)

File: planning/planning_scenarios_distributed_db.adoc (hunk 3 above)
Comment (imcleod) [inline planning/planning_scenarios_distributed_db.adoc]: Reworked with latest patch.

**Pull request merged**